### PR TITLE
Openssl 3.1.4+quic

### PR DESCRIPTION
**Pull request by tmshort**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by tmshort** [comment]
Not worried about Fuzz

**Review by nibanks — APPROVED** [review]

**Comment by richsalz** [comment]
> Not worried about Fuzz

Is this something we *should* care about?  I'm probably fine with either answer, but would like to know why it's an issue.

**Comment by kaduk** [comment]
> > Not worried about Fuzz
> 
> Is this something we _should_ care about? I'm probably fine with either answer, but would like to know why it's an issue.

`clang-15: error: unsupported option '--with-fuzzer-lib=/usr/lib/libFuzzingEngine'` seems to say that the clang available doesn't support libfuzzer.  But that's a pretty modern clang, and libfuzzer is also a LLVM project, so that's kind of weird.  Googling for that flag doesn't find much other than "how to fuzz openssl" postings, but I do wonder if perhaps it needs the C++ driver rather than the C driver.
Anyway, to answer the question, we would probably prefer to have the fuzzers running, but openssl itself is also supposed to be running them, and IIRC we haven't updated the fuzzers' test recipes to attempt to engage any of our QUIC code, so we wouldn't be getting particular benefit from them other than testing stock openssl's code.  So, we should care, but not very urgently.

File: ssl/build.info [hunk 1]
```diff
@@ -37,8 +37,11 @@ IF[{- !$disabled{'deprecated-3.0'} -}]
   SHARED_SOURCE[../libssl]=s3_cbc.c
   SOURCE[../libssl]=ssl_rsa_legacy.c
 ENDIF
-
```
Comment (wbl) [inline ssl/build.info:40]: This one looks a little funny

**Review by wbl — APPROVED** [review]

**Comment by baparham** [comment]
Are tags going to be created for these new versions once they are merged? what about existing missing tags like 3.0.11 that have been merged for a few weeks?

File: ssl/build.info (hunk 1 above)
Comment (tmshort) [reply ssl/build.info:40]: Not incorrect, but strange how it was removed. I can put it back in.

**Comment by tmshort** [comment]
> Are tags going to be created for these new versions once they are merged? what about existing missing tags like 3.0.11 that have been merged for a few weeks?

I will do it soon (possibly this week).

**Pull request merged**